cmake/modules/RootBuildOptions.cmake

```diff
@@ -145,18 +145,18 @@ ROOT_BUILD_OPTION(python3 OFF "Python 3 ROOT bindings, requires python >= 3.1")
 ROOT_BUILD_OPTION(qt OFF "Qt graphics backend, requires libqt >= 4.8")
 ROOT_BUILD_OPTION(qtgsi OFF "GSI's Qt integration, requires libqt >= 4.8")
 ROOT_BUILD_OPTION(r OFF "R ROOT bindings, requires R, Rcpp and RInside")
-ROOT_BUILD_OPTION(rfio ON "RFIO support, requires libshift from CASTOR >= 1.5.2")
+ROOT_BUILD_OPTION(rfio OFF "RFIO support, requires libshift from CASTOR >= 1.5.2")
 ROOT_BUILD_OPTION(roofit OFF "Build the libRooFit advanced fitting package")
 ROOT_BUILD_OPTION(root7 OFF "Build the ROOT 7 interface prototype, requires >= cxx14")
 ROOT_BUILD_OPTION(rpath OFF "Set run-time library load path on executables and shared libraries (at installation area)")
 ROOT_BUILD_OPTION(ruby OFF "Ruby ROOT bindings, requires ruby >= 1.8")
 ROOT_BUILD_OPTION(runtime_cxxmodules OFF "Enable runtime support for C++ modules.")
-ROOT_BUILD_OPTION(sapdb ON "MaxDB/SapDB support, requires libsqlod and libsqlrte")
+ROOT_BUILD_OPTION(sapdb OFF "MaxDB/SapDB support, requires libsqlod and libsqlrte")
 ROOT_BUILD_OPTION(shadowpw ON "Shadow password support")
 ROOT_BUILD_OPTION(shared ON "Use shared 3rd party libraries if possible")
 ROOT_BUILD_OPTION(soversion OFF "Set version number in sonames (recommended)")
```

Comment:
"Recommended", but default to `OFF`... I don't understand.

Reply:
There must be a reason... Reviewers?
As a side note, some comments could be changed, for example the one of cling

Reply:
```
commit 8fe865546e3900231c79b91343d174f0a748a610
Author: Fons Rademakers <Fons.Rademakers@cern.ch>
Date:   Mon Dec 11 16:18:39 2000 +0000

    provide support for shared library version numbers. Use the new configure
    option: --enable-soversion for this. By default shared libs have no version
    number.
```
I think it was just for people doing `-lCore`. We can swap this and see what happens... But I'd recommend to put this in *after* 6.12, in a dedicated commit.

Reply:
Fair enough, this can be changed after the release. One nice thing is that it would help to find out when people link against one version of ROOT then try to run with another. This would make that fail, which is good, since ROOT isn't compatible across versions.

Reply:
FWIW, that's what `TVersionCheck` was invented for, and it works pretty well.